### PR TITLE
Use anonymous ports in S.N.Sockets.Legacy/test/FunctionalTests. 

### PR DESCRIPTION
**Pull request by pgavlin**
This is a follow-up to #3508. With this change, all of the legacy functional
tests use anonymous ports.

This also includes a small change to `SocketPerformanceTests.cs` to address PR feedback from #3508.

### CODE REVIEW

**Comment by pgavlin** [comment]
@cipop @davidsh @stephentoub PTAL

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.Sockets.Legacy/tests/FunctionalTests/AgnosticListenerTest.cs [hunk 1]
```diff
@@ -23,7 +21,7 @@ public AgnosticListenerTest(ITestOutputHelper _log)
         [Fact]
         public void Create_Success()
         {
-            TcpListener listener = TcpListener.Create(TestPortBase);
+            TcpListener listener = TcpListener.Create(0);
```
Comment (CIPop) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/AgnosticListenerTest.cs]: Nit: Please add a comment on what 0 means.

**Comment by CIPop** [comment]
Thanks Pat! I really like how the code ended up looking: this and the previous PR were both very clean changes. 
I already have it in my plan to port the Perf tests back to Desktop. I think this change is also worth porting back to ensure 100% app-compat. I'll add it to my list.

LGTM

**Pull request merged**